src/js/components/TaskView.js

```diff
@@ -60,6 +61,24 @@ class TaskView extends mixin(SaveStateMixin, StoreMixin) {
     super.componentWillMount();
   }
 
+  componentWillReceiveProps(nextProps) {
+    let {filterByStatus, searchString} = this.state;
+    let tasks = this.getFilteredTasks(filterByStatus, nextProps.tasks, searchString);
+    let prevTasks = this.getFilteredTasks(filterByStatus, this.props.tasks, searchString);
+
+    if (!deepEqual(tasks, prevTasks)) {
```

Comment:
Also, for reasons I don't understand, performance on TaskView page is already unbearable when number of tasks is in the order of 100s :/